avocado/plugins/diff.py

```diff
@@ -0,0 +1,427 @@
+                                                    delete=False)
+            tmp_file1.writelines(job1_results)
+            tmp_file1.close()
+
+            prefix = 'avocado_diff_%s_' % job2_id[:7]
+            tmp_file2 = tempfile.NamedTemporaryFile(prefix=prefix,
+                                                    suffix='.txt',
+                                                    delete=False)
+            tmp_file2.writelines(job2_results)
+            tmp_file2.close()
+
+            LOG.info('Job Reports: %s %s' % (tmp_file1.name, tmp_file2.name))
+
+        if (getattr(args, 'open_browser', False) and
+           getattr(args, 'html_output', None) is None):
+
+            prefix = 'avocado_diff_%s_%s_' % (job1_id[:7], job2_id[:7])
+            tmp_file = tempfile.NamedTemporaryFile(prefix=prefix,
+                                                   suffix='.html',
+                                                   delete=False)
+
+            setattr(args, 'html_output', tmp_file.name)
+
+        if getattr(args, 'html_output', None) is not None:
+            self.std_diff_output = False
+            try:
+                html_diff = HtmlDiff()
+                html_diff._legend = """
+                    <table class="diff" summary="Legends">
+                    <tr> <td> <table border="" summary="Colors">
+                    <tr><th> Colors </th> </tr>
+                    <tr><td class="diff_add">&nbsp;Added&nbsp;</td></tr>
+                    <tr><td class="diff_chg">Changed</td> </tr>
+                    <tr><td class="diff_sub">Deleted</td> </tr>
+                    </table></td>
+                    <td> <table border="" summary="Links">
+                    <tr><th colspan="2"> Links </th> </tr>
+                    <tr><td>(f)irst change</td> </tr>
+                    <tr><td>(n)ext change</td> </tr>
+                    <tr><td>(t)op</td> </tr>
+                    </table></td> </tr>
+                    </table>"""
+
+                job_diff_html = html_diff.make_file((_.decode("utf-8")
+                                                    for _ in job1_results),
+                                                    (_.decode("utf-8")
+                                                    for _ in job2_results),
+                                                    fromdesc=job1_id,
+                                                    todesc=job2_id)
+
+                with open(args.html_output, 'w') as f:
+                    f.writelines(job_diff_html.encode("utf-8"))
+
+                LOG.info('HTML diff file: %s' % args.html_output)
+
+            except Exception as e:
+                LOG.error(e)
+                sys.exit(exit_codes.AVOCADO_FAIL)
+
+        if getattr(args, 'open_browser', False):
+            setsid = getattr(os, 'setsid', None)
+            if not setsid:
+                setsid = getattr(os, 'setpgrp', None)
+            inout = file(os.devnull, "r+")
```

Comment:
opening a file using `file` is discouraged in python2.7 and forbidden in python3. Would you please use `open` insteald?